### DISCUSSION TITLE
Typescript type definitions

### DISCUSSION THREAD

**Discussion by kevinleemans** [callstack/react-native-paper#2834] [Q&A]
Is it possible to access the type definitions of the components so that they can be used by typescript users?
I tried finding it myself but can't seem to find it

**Comment by kevinleemans** [comment]
UPDATE: the type definition for components like TextInput are exported but the type definition for Button are not

Workaround is to add export to the type definition inside the component, this is not ideal because every reinstall of the package.json will overwrite these changes

**Reply by Trancever** [reply]
Please use `React.ComponentProps<typeof ComponentFromPaper>` to extract the props type from a component.

**Reply by kevinleemans** [reply]
This works but doesn't provide linting
